src/MongoDB.Driver/TransactionExecutor.cs

```diff
@@ -152,7 +153,9 @@ private static bool CommitWithRetries(IClientSessionHandle clientSession, DateTi
                 catch (Exception ex)
                 {
                     var now = clock.UtcNow; // call UtcNow once since we need to facilitate predictable mocking
-                    if (HasErrorLabel(ex, UnknownTransactionCommitResultLabel) && !HasTimedOut(startTime, now))
+                    if (!IsExceptionExceededTimeLimit(ex) &&
+                        HasErrorLabel(ex, UnknownTransactionCommitResultLabel) &&
+                        !HasTimedOut(startTime, now))
```

Comment:
Replace with:

    if (ShouldRetryCommit(ex, startTime, now))

and add a helper method:

    private static bool ShouldRetryCommit(Exception ex, DateTime startTime, DateTime now)
    {
        return
            HasErrorLabel(ex, UnknownTransactionCommitResultLabel) &&
            !HasTimedOut(startTime, now) &&
            !IsExceededTimeLimitException(ex);
    }